x-pack/plugin/spatial/src/main/java/org/elasticsearch/xpack/spatial/search/aggregations/GeoLineAggregator.java

```diff
@@ -6,6 +6,8 @@
  */
 package org.elasticsearch.xpack.spatial.search.aggregations;
 
+import com.unboundid.util.NotNull;
```

Comment:
I guess if we could avoid using this by just asserting that the input is not null as we are doing in other places.

Reply:
I just checked and we only call these constructors from code that always creates the value source, so no check is needed.

My addition of the null check was as a result of refactoring code that had lots of `if(valueSource==null){}` checks, but that code was supporting both Empty and non-Empty cases, which is no longer the case. So now I just removed the checks.